src/entities/endpoint.entity.ts

```diff
@@ -47,4 +47,11 @@ export class Endpoint extends SharedEntity {
     );
     return this.route;
   }
+
+  /* A hook that runs each time the entity is loaded. It is used to modify the route property of the entity. */
+  @AfterLoad()
+  public DecodeRoute() {
+    this.route = decodeURIComponent(this.route);
+    return this.route;
+  }
```

Comment:
If you decode the route on load, you should check and remove it from every other place it's being decoded